nntrainer/src/pooling2d_layer.cpp

```diff
@@ -33,16 +34,22 @@ int Pooling2DLayer::initialize(bool last) {
   }
 
   this->last_layer = last;
+
   output_dim.batch(input_dim.batch());
   output_dim.channel(input_dim.channel());
-  output_dim.height(
-    (input_dim.height() - pooling_size[0] + 2 * padding[0]) / stride[0] + 1);
-  output_dim.width(
-    (input_dim.width() - pooling_size[1] + 2 * padding[1]) / stride[1] + 1);
+
+  if (pooling_type == PoolingType::max ||
+      pooling_type == PoolingType::average) {
+    output_dim.height(
+      (input_dim.height() - pooling_size[0] + 2 * padding[0]) / stride[0] + 1);
+    output_dim.width(
+      (input_dim.width() - pooling_size[1] + 2 * padding[1]) / stride[1] + 1);
+  }
 
```

Comment:
is `output_dim` == `batch:channel:1:1` guaranteed for global max / global average?

Reply:
As far as I know. yes.